enhancements/service-catalog/service-catalog-and-broker-removal.md

```diff
@@ -0,0 +1,211 @@
+clusters.
+
+## Motivation
+
+This is the next step in the removal of the Service Catalog and Brokers.
+
+* Deprecated APIs in OpenShift 4.1 [release
+  notes](https://docs.openshift.com/container-platform/4.1/release_notes/ocp-4-1-release-notes.html).
+* In OpenShift 4.2, a more detailed annoucement of their deprecation was
+  released: [4.2 release notes](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html#ocp-41-deprecated-features)
+* [Support for templates](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html#ocp-4-2-general-web-console-updates) without using the Template Service Broker and Service
+  Catalog added to OpenShift 4.2.
+* Alerts were added to the Service Catalog operators:
+  * https://github.com/openshift/cluster-svcat-apiserver-operator/blob/master/manifests/0000_90_cluster-svcat-apiserver-operator_02-operator-servicemonitor.yaml#L28-L45
+  * https://github.com/openshift/cluster-svcat-controller-manager-operator/blob/master/manifests/0000_90_cluster-svcat-controller-manager-operator_02_servicemonitor.yaml#L27-L44
+* Alerts were added to the Ansible Service Broker operator:
+  * https://github.com/openshift/ansible-service-broker/blob/master/operator/roles/ansible-service-broker/templates/broker.service-monitor.yaml.j2
+* Alerts were added to the Template Service Broker operator:
+  * https://github.com/openshift/template-service-broker-operator/blob/master/roles/template-service-broker/templates/tsb.prometheus-alert.yaml
+
+### Goals
+
+1. Implement a mechanism that ensures the Service Catalog and Service Brokers
+   are not present after an upgrade to OpenShift 4.4. (ENG)
+1. Prevent users from upgrading to OpenShift 4.4 if the Service Catalog or the
+   Service Brokers are in use. (ENG)
+1. Remove the Service Catalog from the OpenShift 4.4 release payload (ART)
+   * Will have to keep the Service Catalog operators if we choose to repurpose
+     them.
+1. Remove the Ansible Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from productization (ART)
+1. Remove the Ansible Service Broker from productization (ART)
+1. Remove the Template Service Broker Operator from productization (ART)
+1. Remove the Ansible Service Broker Operator from productization (ART)
+1. Remove the Service Catalog from productization (ART)
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Implement a mechanism that ensures the Service Catalog and Service Brokers are
+not present after an upgrade to OpenShift 4.4.
+
+We could potentially block upgrades from happening until the admin removes the
+Service Catalog and the brokers. This would need a mechanism that blocks
+upgrades. There is an `Upgradeable: False` in the operator's status but not sure
+if that is checked by the CVO during upgrades.
```

Comment:
CVO should be checking this since openshift/cluster-version-operator#243

Reply:
@wking if I set `Upgradeable` to false, will that prevent major upgrades, i.e. 4.3 -> 4.4? What about z-stream, i.e. 4.3.0 -> 4.3.1? Presumably I'd like to allow upgrades in the z-stream, but not going to 4.4.